lib/packages.ts

```diff
@@ -100,23 +100,32 @@ function loadPackageJson(p: string) {
   return pkg;
 }
 
-function _findAllPackageJson(dir: string, exclude: RegExp): string[] {
-  const result: string[] = [];
-  fs.readdirSync(dir).forEach((fileName) => {
-    const p = path.join(dir, fileName);
+function* _findAllPackageJson(dir: string, exclude: RegExp): Iterable<string> {
```

Comment:
Optional nit: Do you want to rename this to something more specific now? e.g. `_findPrimaryPackageJsonFiles()`